rust/sdk-processor/src/steps/common/processor_status_saver.rs

```diff
@@ -0,0 +1,144 @@
+    ) -> Result<(), ProcessorError> {
+        let end_timestamp = last_success_batch
+            .metadata
+            .end_transaction_timestamp
+            .as_ref()
+            .map(|t| parse_timestamp(t, last_success_batch.metadata.end_version as i64))
+            .map(|t| t.naive_utc());
+        match self {
+            ProcessorStatusSaverEnum::Default {
+                conn_pool,
+                processor_name,
+            } => {
+                let status = ProcessorStatus {
+                    processor: processor_name.clone(),
+                    last_success_version: last_success_batch.metadata.end_version as i64,
+                    last_transaction_timestamp: end_timestamp,
+                };
+
+                // Save regular processor status to the database
+                execute_with_better_error(
+                    conn_pool.clone(),
+                    diesel::insert_into(processor_status::table)
+                        .values(&status)
+                        .on_conflict(processor_status::processor)
+                        .do_update()
+                        .set((
+                            processor_status::last_success_version
+                                .eq(excluded(processor_status::last_success_version)),
+                            processor_status::last_updated.eq(excluded(processor_status::last_updated)),
+                            processor_status::last_transaction_timestamp
+                                .eq(excluded(processor_status::last_transaction_timestamp)),
+                        )),
+                    Some(" WHERE processor_status.last_success_version <= EXCLUDED.last_success_version "),
+                )
+                .await?;
+
+                Ok(())
+            },
+            ProcessorStatusSaverEnum::Backfill {
+                conn_pool,
+                backfill_alias,
+                backfill_start_version,
+                backfill_end_version,
+            } => {
+                let lst_success_version = last_success_batch.metadata.end_version as i64;
+                let backfill_status = if backfill_end_version.is_some_and(|backfill_end_version| {
+                    lst_success_version >= backfill_end_version as i64
+                }) {
+                    BackfillStatus::Complete
+                } else {
+                    BackfillStatus::InProgress
+                };
+                let status = BackfillProcessorStatus {
+                    backfill_alias: backfill_alias.clone(),
+                    backfill_status,
+                    last_success_version: lst_success_version,
+                    last_transaction_timestamp: end_timestamp,
+                    backfill_start_version: backfill_start_version.unwrap_or(0) as i64,
+
+                    backfill_end_version: backfill_end_version
+                        .unwrap_or(last_success_batch.metadata.end_version)
+                        as i64,
```

Comment:
good call out. This way, it'll be clear that the BFP will run forever rather than incorrectly indicating that the `last_processed_version` is equal to the `backfill_end_version` and status being `in_progress`.